src/segment_owm.go

```diff
@@ -61,22 +65,58 @@ func (d *owm) string() string {
 	return text
 }
 
-func (d *owm) setStatus() error {
+func (d *owm) getResult() (*OWMDataResponse, error) {
 	apikey := d.props.getString(APIKEY, ".")
 	location := d.props.getString(LOCATION, "De Bilt,NL")
 	units := d.props.getString(UNITS, "standard")
 	timeout := d.props.getInt(HTTPTimeout, DefaultHTTPTimeout)
+	cachefile := d.props.getString(CACHEFILE, "")
+
+	if cachefile != "" {
+		stats, err := os.Stat(cachefile)
+		if err == nil {
+			if time.Since(stats.ModTime()).Minutes() < 10 {
+				cachedData, err := os.ReadFile(cachefile)
+				if err == nil {
+					q := new(OWMDataResponse)
+					err := json.Unmarshal(cachedData, q)
+					if err == nil {
+						return q, nil
+					}
+				}
+			}
+		}
+	}
 
 	url := fmt.Sprintf("http://api.openweathermap.org/data/2.5/weather?q=%s&units=%s&appid=%s", location, units, apikey)
 	body, err := d.env.doGet(url, timeout)
 	if err != nil {
-		return err
+		return new(OWMDataResponse), err
 	}
 	q := new(OWMDataResponse)
 	err = json.Unmarshal(body, &q)
+	if err != nil {
+		return new(OWMDataResponse), err
+	}
+
+	if cachefile != "" {
+		err := os.WriteFile(cachefile, body, 0600)
+		if err != nil {
```

Comment:
When there's an error with the cache, I don't see why we do not return the data.
I would make use of `defer` to write the cache after the entire logic executes.